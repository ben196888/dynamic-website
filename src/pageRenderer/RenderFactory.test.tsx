import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import factory from './RenderFactory';

describe('RenderFactory', () => {
  test('render string', () => {
    const actual = factory('John Doe', {});
    expect(typeof actual).toBe('string');
    expect(actual).toBe('John Doe');
  });
  test.each([
    ['div'],
    ['a'],
    ['button'],
    ['span'],
    ['h1'],
  ])('render basic html component - %s', (type) => {
    const cmp = factory({ key: `unique-key-${type}`, type }, {});
    if (typeof cmp !== 'string') {
      const actual = renderer.create(cmp).toJSON();
      expect(actual).toMatchSnapshot();
    }
  });
  test('render my component', () => {
    const cmp = factory({ key: 'unique-key-mycomponent', type: 'MyComponent' }, {});
    if (typeof cmp !== 'string') {
      const actual = shallow(cmp);
      expect(actual.find('MyComponent')).toBeDefined();
    }
  });
});
