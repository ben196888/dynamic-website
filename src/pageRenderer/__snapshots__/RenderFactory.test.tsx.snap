// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderFactory render basic html component - a 1`] = `<a />`;

exports[`RenderFactory render basic html component - button 1`] = `<button />`;

exports[`RenderFactory render basic html component - div 1`] = `<div />`;

exports[`RenderFactory render basic html component - h1 1`] = `<h1 />`;

exports[`RenderFactory render basic html component - span 1`] = `<span />`;
